src/main/java/ai/shreds/adapter/AdapterPurchaseController.java

```diff
@@ -0,0 +1,49 @@
+package ai.shreds.adapter;
+
+import ai.shreds.adapter.AdapterPurchaseRequestDTO;
+import ai.shreds.adapter.AdapterPurchaseResponseDTO;
+import ai.shreds.application.ApplicationPurchaseServiceInputPort;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import javax.validation.Valid;
+
+@Slf4j
+@RestController
+@RequestMapping("/purchase")
+@RequiredArgsConstructor
+public class AdapterPurchaseController {
+
+    private final ApplicationPurchaseServiceInputPort purchaseService;
```

Comment:
Use ApplicationSchoolInfoInputPort instead of ApplicationPurchaseServiceInputPort